Source/WebCore/dom/Document.cpp

```diff
@@ -3562,7 +3562,7 @@ const URL& Document::urlForBindings() const
             return false;
 
         auto* topDocumentLoader = topDocument().loader();
-        if (!topDocumentLoader || !topDocumentLoader->networkConnectionIntegrityPolicy().contains(NetworkConnectionIntegrity::Enabled))
+        if (!topDocumentLoader || !topDocumentLoader->originalNetworkConnectionIntegrityPolicy().contains(NetworkConnectionIntegrity::Enabled))
```

Comment:
Hm... I wonder if the `original` part of `originalNetworkConnectionIntegrityPolicy` might be a bit ambiguous (namely, the fact that "original" here refers to the loader that originated the current load).

Perhaps, something along the lines of `originatorNetworkConnectionIntegrityPolicy` or `referrerNetworkConnectionIntegrityPolicy`?

Otherwise, this seems reasonable — but someone more familiar with loading code should probably review (e.g. @cdumez  or @achristensen07).